client/lib/resize-image-url/index.js

```diff
@@ -104,23 +107,22 @@ export default function resizeImageUrl( imageUrl, resize, height, makeSafe = tru
 	}
 
 	// Map sizing parameters, multiplying their values by the scale factor
-	assign(
-		parsedUrl.query,
-		mapValues( resize, ( value, key ) => {
-			if ( 'resize' === key || 'fit' === key ) {
-				return value
-					.split( ',' )
-					.map( scaleByFactor )
-					.join( ',' );
-			} else if ( includes( SIZE_PARAMS, key ) ) {
-				return scaleByFactor( value );
-			}
-
-			return value;
-		} )
-	);
+	const mapped = mapValues( resize, ( value, key ) => {
+		if ( 'resize' === key || 'fit' === key ) {
+			return value
+				.split( ',' )
+				.map( scaleByFactor )
+				.join( ',' );
+		} else if ( SIZE_PARAMS.includes( key ) ) {
+			return scaleByFactor( value );
+		}
 
-	delete parsedUrl.search;
+		return value;
+	} );
+
+	for ( const key of Object.keys( mapped ) ) {
+		parsedUrl.searchParams.set( key, mapped[ key ] );
+	}
```

Comment:
same idea as above with mutation. personally I end up with more confidence and less anxiety if I can see that a value is set once and I don't have to hunt the rest of the function to see if it's changed later, or could be changed in different ways depending on runtime characteristics.

```js
const resizedUrl = new URL( withoutSizes.href );
Object.keys( mapped ).forEach( key => resizedUrl.searchParams.set( key, mapped[ key ] ) );
```

at which point we may think about creating a new name altogether for what we're doing. it takes me a second glance to understand what we're trying to do here

```js
const withSearchParams = ( baseUrl, params ) => {
	const url = new URL( baseUrl.href );
	Object.keys( params ).forEach( name => url.searchParams.set( name, params[ name ] ) );
	return url;
}

const withoutSearchParams = ( baseUrl, paramNames ) => {
	const url = new URL( baseUrl.href );
	paramNames.forEach( name => url.searchParams.delete( name ) );
	return url;
}
```

```js
const withoutSizes = withoutSearchParams( parsedUrl, SIZE_PARAMS );
```

```js
return withSearchParams( withoutSizes, mapped ).href;
```

`mapped` may not be the most ideal name either, but I'm not trying to push too far into this specific module, just suggesting that having those names might more quickly indicate the intention of the loops and mutation, as well as give us a more concrete unit to test.

Reply:
Again, I think it's pretty idiomatic to create a result value and mutate it as the function progresses. Hopefully renaming the variable made it clearer, but let me know if you'd still like me to use different `const`s for each individual stage of processing.

Reply:
this shouldn't block the PR